### ISSUE TITLE
Error 80004005 in classic ASP

### ISSUE THREAD

**Issue by Ithier-wedia** [SpiderLabs/ModSecurity#613]
When I try to read a non empty request parameter with method POST, in my ASP application, I always have the error:

```
erreur '80004005'
/test.asp, ligne 4
```

I can avoid this error by setting the SecStreamInBodyInspection to On, but it is not a very useful solution as it slow down a lot the uploads.

Here is a test page

```
<html>                    
    <head></head>                                    
    <body>                                                 
        Read : <%=Request("test")%>                                                       
        <form method="post">                                                                          
            <input type="text" name="test">                                                              
        </form>                                    
    </body>         
</html>
```

If you load this page it works fine.
If you enter no value in the input and press enter, it still works fine.
If you enter a value in the input and press enter, you will have the error.

If you modify the form method from POST to GET, no more error.

Environment :
Windows 8, IIS8, ASP 32 bits

**Comment by victorhora** [comment]
Apologies for the lack of response :(

Please let us know if the issue persists with a newer version of ModSecurity such as 2.9.2 or v3 ([libModSecurity](https://github.com/SpiderLabs/ModSecurity/tree/v3/master))

Thanks!

**Comment by beatinger** [comment]
Hello Victor,

I would like to CONFIRM with you that this issue still exists exactly the same as described above, even whilst using v2.9.2.  When I uninstall ModSecurity, the issue goes away.  I will try v3 next, and let you know what happens by coming back here and adding another note.  Please let me know when this will be fixed, as I desperately need to block Chinese IP addresses from accessing my website.  Thank you!

**Comment by beatinger** [comment]
Hello Victor,

Actually I just found that v3 is not available as an MSI for installation on Windows (at: http://www.modsecurity.org/download.html).  Please let me know when you think there might be a fix in place.  I really need a way to stop IP addresses from China from accessing my servers.  I suppose I could add the PeerBlock firewall, but don't really want to go that route.  Thank you very much for your time!

**Comment by victorhora** [comment]
Still could not reproduce the issue on a recent IIS build for 2.9.x.

![image](https://user-images.githubusercontent.com/15767386/48739563-5c840800-ec22-11e8-8cbd-dfd53980d321.png)

```
2018-11-19 22:28:02 127.0.0.1 POST /issue-613.asp - 80 - 127.0.0.1 Mozilla/5.0+(Windows+NT+6.1;+WOW64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/70.0.3538.102+Safari/537.36 200 0 0 916
```

Please bear in mind that the for the IIS build, we currently recommend setting [SecStreamInBodyInspection](https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual-%28v2.x%29#SecStreamInBodyInspection) to ```On```. (See #1299).

The problem with the slowdown with the uploads is not present anymore since https://github.com/SpiderLabs/ModSecurity/commit/afae6906552fa5852ab295920e25da35392c8443 (issue #1366), so it should be safe to add this for most environments.

**Comment by victorhora** [comment]
There's a chance that the behaviour @beatinger is experiencing on the newer version could be related with a false positive. To be certain, check the EventViewer / audit_logs and debug_logs. 

If certain is not a false positive, please provide a detailed description of the running environment (IIS plugins, versions, customization, ruleset) and preferably with reproducible steps leading to the issue alongside the logs so we investigate properly and reopen the issue if still a bug.

Thanks.

**Comment by beatinger** [comment]
Thank you for responding.  This is a clean and fresh IIS install on Windows server 2012.  The issue has to do strictly with the instantiation of objects, such as a COM object or SQL server object.  In fact, the following code (contained in the GLOBAL.ASA file), will fail immediately after restarting IIS (iisrestart):  

<object runat="server" scope="session" id="oBrowserType" progid="MSWC.BrowserType"></object>
sBrowserName = oBrowserType.Browser

Immediately after uninstalling ModSecurity, everything works perfectly again.  I have not tried what you mentioned above ("setting SecStreamInBodyInspection to ON"), but can do that, and get back to you.  Can you please let me know in which file exactly that this setting is made, and where that file is?  Thank you very much!

**Comment by beatinger** [comment]
Unfortunately, something in this editor, stripped out the HTML code.  I am going to try to include it again here, but will remove the less than/greater than brackets, and use apostrophes instead, as follows:

'object runat="server" scope="session" id="oBrowserType" progid="MSWC.BrowserType"'

'/object'

**Comment by RosscoeP** [comment]
I had similar issues. 

Windows Web Server 2008 R2, IIS 7.5 and ModSecurity 2.9.1.

I found the ASP request object had TotalBytes that made sense, but BinaryRead(request.TotalBytes) returned 80004005. What was occurring was that ModSecurity was reading (and emptying the return buffer) but leaving all other properties intact.

Therefore it may be that the SecStreamInBodyInspection change will fix this.

**Comment by beatinger** [comment]
Where is that SecStreamInBodyInspection setting at?

**Comment by victorhora** [comment]
@beatinger 

This issue was already fix on the 2.9.3 release. You may consider upgrading.

If you want to change it manually on your local installation, search for a file named _modsecurity.conf_ on your disk. If I remember correctly, by default it will be located on the folder where your ModSecurityIIS package is installed.
 https://github.com/SpiderLabs/ModSecurity/commit/45337265f1df8f45908390caf896158d241aae72

**Comment by beatinger** [comment]
I installed PeerBlock and paid for the yearly subscription service to obtain daily updates to block lists.  It seems to be working well, but there are so many blocks going on, that it is slowing down my server.  I really need to block all of this nonsense in my firewall, which is a Cisco ASA 5450, but I do not know how.
